Mathlib/GroupTheory/SpecificGroups/Cyclic.lean

```diff
@@ -117,6 +117,15 @@ lemma isCyclic_iff_exists_orderOf_eq_natCard [Finite α] :
     IsCyclic α ↔ ∃ g : α, orderOf g = Nat.card α := by
   simp_rw [isCyclic_iff_exists_zpowers_eq_top, ← card_eq_iff_eq_top, Nat.card_zpowers]
 
+@[to_additive]
+lemma isCyclic_iff_exists_natCard_le_orderOf [Finite α] :
+    IsCyclic α ↔ ∃ g : α, Nat.card α ≤ orderOf g := by
+  obtain ⟨⟩ := nonempty_fintype α
+  rw [isCyclic_iff_exists_orderOf_eq_natCard, Nat.card_eq_fintype_card]
+  apply exists_congr
+  intro g
+  exact ⟨Eq.ge, le_antisymm orderOf_le_card_univ⟩
```

Comment:
Can you add a theorem `orderOf_le_card` stating `orderOf g ≤ Nat.card α`? We already have the more specific `Subgroup.orderOf_le_card`.

Reply:
Done